### DISCUSSION TITLE
where to get driver of Feather M4 CAN SuperCAN?

### DISCUSSION THREAD

**Discussion by kiddlotw** [jgressmann/supercan#10] [General]
Hi, there
I flashed supercan firmware (new release v1.0.16) on my Adafruit Feather M4 CAN Express,
and then I ran "supercan_inst.exe" to install driver on windows 10.
however, my computer still cannot recognize it
only I can see is "Feather M4 CAN SuperCAN (cc95f3e0)" with unknown state in device manager

double-click to get detail:
The drivers for this device are not installed. (Code 28)
There are no compatible drivers for this device.

![image](https://user-images.githubusercontent.com/107239551/173005145-fa1795d1-2db6-4984-ad45-270ffb247699.png)

**Comment by jgressmann** [comment]
I really must test more on Windows. Please install [the latest firmware](https://github.com/jgressmann/supercan/releases/tag/latest-master) and try again.

**Reply by kiddlotw** [reply]
> I really must test more on Windows. Please install [the latest firmware](https://github.com/jgressmann/supercan/releases/tag/latest-master) and try again.

thanks for so quick response, but I still got unkown device in windows 10,
as I saw during the installation (supercan_inst.exe), it didnt say anything about driver installation, just verified vc++ 2015-2019 redist

then, I followed some steps in "D5035-01 Firmware Update Guide for Windows", to install WinUSB driver by using Zadig
![install_winusb6_drv](https://user-images.githubusercontent.com/107239551/173051553-2f8373bb-6df6-4102-a56b-95d2197f9043.png)

and, I tried supercan_app64.exe in c:\Program Files\SuperCAN\bin, however, I got the message
![run_supercan_app64](https://user-images.githubusercontent.com/107239551/173051900-01bfadcd-f0c9-4455-89e4-deb80ec72f91.png)

any idea? please let me know, I can try. :-)

**Reply by jgressmann** [reply]
Unless the device is functioning properly (verify in device manager) you won't be able to connect.

**Comment by jgressmann** [comment]
You shouldn't have to use Zadig anymore, it was only necessary in the past prior to MS OS 2.0 descriptors support in the firmware.

Please try to delete / uninstall the device in the device manager. With the device plugged in, select the device, the either press delete or use the context menu to uninstall the device. Once uninstalled, replug the device into the computer. The device should come up without error.

**Reply by kiddlotw** [reply]
yes, you're right, using Zadig is not correct
with the firmware of Jun.10, my windows 10 computer really can get M4 express device ready without any operation
![M4_SuperCan_ch0_title](https://user-images.githubusercontent.com/107239551/173090784-663c5960-a418-4694-8829-9c926935b071.png)

more, I test it with can++ application, finally..... selecting SuperCan becomes reality
![select_superCan](https://user-images.githubusercontent.com/107239551/173091402-b5aac798-3a4a-4049-943d-cdbbf9bab2b2.png)

like amazing charm!!  thank you, bro~
